pkg/server/preprocessor/auth.go

```diff
@@ -113,9 +114,18 @@ func (p *UserAuthenticator) Preprocess(payload *router.Payload, response *router
 	// For master access key, it is possible to impersonate any user of
 	// the caller's choosing.
 	if payload.HasMasterKey() {
-		if userID, ok := payload.Data["_user_id"].(string); ok {
-			payload.UserInfoID = userID
-			payload.Context = context.WithValue(payload.Context, router.UserIDContextKey, userID)
+		userID, ok := payload.Data["_user_id"].(string)
+		if !ok {
```

Comment:
This wouldn’t work as I would expect in these situations:

* If payload contains `_user_id` key and it is an empty string (`userID` is empty string and `ok` is `true`)
* If payload contains `_user_id` key and it is something that is not a string (integer or map), should the server ignores the input and assume `_god` in this situation?